### ISSUE TITLE
Word2Vec Iterations and Phrases

### ISSUE THREAD

**Issue by quhfus** [piskvorky/gensim#467]
When learning a word2vec model (e.g. on Wikipedia), how many iteration are recommended? Is one iteration enough? Does this depend on the given data set? This brings me to another question related to gensim. I recently learned a word2vec model by using: 

    model = gensim.models.Word2Vec(sentences=corpus, size=500, window=8, min_count=3, workers=25, sample=1e-4, iter=5)

However, there was only one iteration from 0 to 100% during the learning phase. The "iter=5" parameter should be correct. Is this a problem on my side?

My second question is about phrases. In the original c word2vec code, when using vectors-phrase.bin, the implementation learns a word2vec model containing not only words but also some phrases (e.g. san_francisco). As far as I understand the procedure, the phrases within the sentences are learned first and afterwards the word2vec model is learned with these phrases.
When I want to do the same with the gensim implementation, what do I have to do? 

I guess the following:
1. using gensim.model.Phrases(sentences) to create a phrase model.
2. replacing all bigrams, trigrams,... of my word2vec corpus with those learned in procedure 1.
3. learn the word2vec model

Basically, this is not too complicated but does gensim offer a function which combines these steps?

**Comment by gojomo** [comment]
For iterations, there's no right answer: whatever works with your data. Smaller datasets are more likely to benefit from more iterations. 

The percentage completion estimate is based on the known total amount of training (including all iterations). Did the sentence/example counts shown at the end match 5x your original dataset? 

Yes, you must apply the Phrases steps yourself, then send the transformed corpus to Word2Vec. 

In the future, it's better to ask such usage questions (not involving a certain bug or new-feature-request) on the project discussion forum: https://groups.google.com/forum/#!forum/gensim

**Comment by quhfus** [comment]
Thank you very much.
@Iteration Ok. That sounds logical to me. I thought for each epoch from 1 to 100 percent. 
@Usage_questions All right. Thanks for the link.
